File: app-java/src/main/java/net/mEmoZz/yts/java/ui/main/MainView.java
package net.mEmoZz.yts.java.ui.main;

import java.util.List;
import net.mEmoZz.yts.java.data.models.BaseMovie;
import net.mEmoZz.yts.java.ui.base.BaseView;

/**
 * Authored by Mohamed Fathy on 08 Mar, 2018.
 * Contact: muhamed.gendy@gmail.com
 */

interface MainView extends BaseView {

  void setupRecycler();

  void setupRefreshLayout();

  void setRecyclerAdapter(List<BaseMovie.Movie> moviesList, boolean refresh);

  void onLoadMore(int currentPage);

  void enableRefreshLayout();

  void showRefreshLayout();

  void hideRefreshLayout();

  void showLoadingBar();

  void hideLoadingBar();

  void showRecycler();

  void hideRecycler();

  void showNoConnectionView();

  void showEmptyLisView();

  void showErrorView();

  void hideHolderViews();
}

File: app-kotlin/src/main/java/net/mEmoZz/yts/kotlin/ui/main/MainView.kt
package net.mEmoZz.yts.kotlin.ui.main

import net.mEmoZz.yts.kotlin.data.models.BaseMovie
import net.mEmoZz.yts.kotlin.ui.base.BaseView

/**
 * Authored by Mohamed Fathy on 08 Mar, 2018.
 * Contact: muhamed.gendy@gmail.com
 */

internal interface MainView : BaseView {

  fun setupRecycler()

  fun setupRefreshLayout()

  fun setRecyclerAdapter(moviesList: List<BaseMovie.Movie>, refresh: Boolean)

  fun onLoadMore(currentPage: Int)

  fun enableRefreshLayout()

  fun showRefreshLayout()

  fun hideRefreshLayout()

  fun showLoadingBar()

  fun hideLoadingBar()

  fun showRecycler()

  fun hideRecycler()

  fun showNoConnectionView()

  fun showEmptyLisView()

  fun showErrorView()

  fun hideHolderViews()
}

